app/assets/javascripts/player.js

```diff
@@ -1,3 +1,122 @@
+var radioCategory;
+
+// ##### HELPER FUNCTIONS #####
+// Toggle play and pause buttons
+function toggleButton(track) {
+  if (track.paused) {
+    $('#play-pause-btn').attr('src', '/assets/play.png');
+  } else {
+    $('#play-pause-btn').attr('src', '/assets/pause.png');
+  }
+}
+
+// Initialize the user on soundcloud
+function initialize(client) {
+  SC.initialize({
+    client_id: client
+  });
+}
+
+// Stops and destroy the tracks
+function stopMusic(track) {
+  if(typeof(soundManager) !== 'undefined'){
+    soundManager.stopAll();
+    var soundsIDs = soundManager.soundIDs;
+    // var trackID = soundsIDs[0];
+    for (i = 0; i < soundsIDs.length; i++) {
+      soundManager.destroySound(soundsIDs[i]);
+    }
+  }
+}
+
+// PLAY A RADIO
+$('#radios-play-list').on('click', '.radios', function(e) {
+  e.preventDefault();
+  radioName = $(this).attr('name');
+  radioCategory = $(this).attr('category');
+  $('#radio-playing').text(radioName);
+  getTracks();
+});
+
+// CREATES THE PLAYLIST
+function getParameters(radio) {
+  var serchParameters;
+  if (radio === 'genres') {
+    var serchParameters = { state: 'finished', sharing: 'public', streamable: true, genres: radioName, limit: 200 };
+  }else{
+    var serchParameters = { state: 'finished', sharing: 'public', streamable: true, q: radioName, limit: 200 };
+  }
+  return serchParameters
+}
+
+function getTracks() {
+  initialize(CLIENT_ID);
+  SC.get('/tracks', getParameters(radioCategory), function(songs) {
+    stopMusic();
+    currentTracks = songs;
+    playSong(currentTracks);
+  });
+}
+
+// GETS THE SONG AND PLAYS IT
+function playSong(tracks) {
+  var i = Math.floor(Math.random() * tracks.length);
+  $('#song-title').replaceWith('<p id="song-title"><marquee behavior="scroll" direction="left">' + tracks[i].title + '</marquee></p>');
+  SC.stream('/tracks/' + tracks[i].id, { flashVersion: 9, autoPlay: true, multiShot: false, onfinish: function() {
+    stopMusic(); playSong(currentTracks); } }, function(track) {
+    songController(track);
+  });
+}
+
+// NEXT SONG
+$('#next-btn').on('mousedown', function() {
+  var element = $(this).attr('id');
+  // animateButtons(element);
+  stopMusic();
+  playSong(currentTracks);
+});
+
+// CONTROLLER FOR THE PLAY PAUSE BUTTON
+function songController(track) {
+  toggleButton(track);
+  timer(track);
+
+  $('#play-pause-btn').on('mousedown', function() {
+    var element = $(this).attr('id');
+    animateButtons(element)
+  });
+
+  $('#play-pause-btn').on('click', function() {
+    if (track.paused) {
+      track.play();
+    } else {
+      track.pause();
+    }
+    toggleButton(track);
+  });
+}
+
+// UPDATES THE SONG TIMER
+function timer(track) {
+  setInterval(function() {
+    if (track.sID === soundManager.soundIDs[0]) {
+        var seconds = track.position / 1000;
+        min = Math.floor(seconds / 60);
+        sec = Math.floor(seconds % 60);
+        min.toString().length === 1 ? sMinutes = '0' + min : sMinutes = min;
```

Comment:
Expected an assignment or function call and instead saw an expression.<br>'min' is not defined.<br>'sMinutes' is not defined.